frontend/src/static/js/v_summary.js

```diff
@@ -362,33 +361,88 @@ window.vSummary = {
       return null;
     },
     sortFiltered() {
-      const full = [];
-      if (!this.filterGroupRepos) {
-        full.push([]);
-      }
-
-      this.filtered.forEach((users) => {
-        if (this.filterGroupRepos) {
-          users.sort(comparator((ele) => ele[this.filterSort]));
-          full.push(users);
-        } else {
-          users.forEach((user) => full[0].push(user));
-        }
-      });
-
-      if (!this.filterGroupRepos) {
-        full[0].sort(comparator((ele) => {
-          const field = ele[this.filterSort];
-          return field.toLowerCase ? field.toLowerCase() : field;
-        }));
+      let full = [];
+      if (this.filterGroupSelection === 'groupByNone') {
+        full[0] = this.filterByNone(this.filtered);
+      } else if (this.filterGroupSelection === 'groupByAuthors') {
+        full = this.filterByAuthors(this.filtered);
+      } else {
+        full = this.filterByRepos(this.filtered);
       }
 
       if (this.filterSortReverse) {
         full.forEach((repo) => repo.reverse());
+        if (this.filterGroupSelection === 'groupByAuthors') {
+          full.reverse();
+        }
       }
 
       this.filtered = full;
     },
+
+    filterByRepos(repos) {
+      const sortedRepos = [];
+      repos.forEach((users) => {
+        users.sort(comparator((ele) => ele[this.filterSort]));
+        sortedRepos.push(users);
+      });
+      return sortedRepos;
+    },
+    filterByNone(repos) {
+      const sortedRepos = [];
+      repos.forEach((users) => {
+        users.forEach((user) => {
+          sortedRepos.push(user);
+        });
+      });
+      sortedRepos.sort(comparator((ele) => {
+        const field = ele[this.filterSort];
+        return field.toLowerCase ? field.toLowerCase() : field;
+      }));
+      return sortedRepos;
+    },
+    filterByAuthors(repos) {
+      const authorMap = {};
+      const authorData = {};
+      const filtered = [];
+      repos.forEach((users) => {
+        users.forEach((user) => {
+          if (Object.keys(authorMap).includes(user.name)) {
+            authorMap[user.name].push(user);
+          } else {
+            authorMap[user.name] = [user];
+          }
+        });
+      });
+      // If filterSort is related to variance & contribution, sum them up
+      if (repos[0] && !Number.isNaN(repos[0][0][this.filterSort])) {
+        Object.keys(authorMap).forEach((author) => {
+          authorData[author] = {};
+          authorData[author][this.filterSort] = [0, 0];
+          authorMap[author].forEach((repo) => {
+            authorData[author][this.filterSort][0] += 1;
+            authorData[author][this.filterSort][1] += repo[this.filterSort];
```

Comment:
what does this `[0,0]` do? is it total commits VS num of repo? can add a comment? `// [numRepos, numCommits]`